OMEdit/OMEditLIB/Modeling/Model.cpp

```diff
@@ -1426,26 +1450,66 @@ namespace ModelInstance
       modifierValue = mpModel->getModifier().getModifierValue(modifierNames);
       // Case 3
       if (modifierValue.isEmpty()) {
-        modifierValue = Element::getModifierValueFromInheritedType(mpModel, modifierNames);
+        modifierValue = Component::getModifierValueFromInheritedType(mpModel, modifierNames);
       }
     }
     return modifierValue;
   }
 
-  QString Element::getModifierValueFromInheritedType(Model *pModel, QStringList modifierNames)
+  QString Component::getModifierValueFromInheritedType(Model *pModel, QStringList modifierNames)
   {
     QString modifierValue = "";
-    foreach (auto pExtend, pModel->getExtends()) {
-      modifierValue = pExtend->getModifier().getModifierValue(modifierNames);
-      if (modifierValue.isEmpty()) {
-        modifierValue = Element::getModifierValueFromInheritedType(pExtend, modifierNames);
-      } else {
-        return modifierValue;
+    foreach (auto pElement, pModel->getElements()) {
+      if (pElement->isExtend() && pElement->getModel()) {
+        auto pExtend = dynamic_cast<Extend*>(pElement);
+        modifierValue = pExtend->getModel()->getModifier().getModifierValue(modifierNames);
+        if (modifierValue.isEmpty()) {
+          modifierValue = Component::getModifierValueFromInheritedType(pExtend->getModel(), modifierNames);
+        } else {
+          return modifierValue;
+        }
       }
     }
     return modifierValue;
   }
 
+  bool Component::isComponent() const
+  {
+    return mKind.compare(QStringLiteral("component")) == 0;
+  }
+
+  Extend::Extend(Model *pParentModel, const QJsonObject &jsonObject)
+    : Element(pParentModel)
+  {
+    mKind = "";
+    mpExtendsAnnotation = std::make_unique<Annotation>(pParentModel);
+    deserialize(jsonObject);
+  }
+
+  void Extend::deserialize(const QJsonObject &jsonObject)
+  {
+    if (jsonObject.contains("$kind")) {
+      mKind = jsonObject.value("$kind").toString();
+    }
+
+    if (jsonObject.contains("modifiers")) {
+      mExtendsModifier.deserialize(jsonObject.value("modifiers"));
+    }
+
+    if (jsonObject.contains("annotation")) {
+      mpExtendsAnnotation->deserialize(jsonObject.value("annotation").toObject());
+    }
+
+    if (jsonObject.contains("baseClass")) {
+      mpModel = new Model(jsonObject.value("baseClass").toObject(), this);
+    }
+  }
+
+  bool Extend::isExtend() const
+  {
+    return mKind.compare(QStringLiteral("extends")) == 0;
```

Comment:
This seems a bit odd. An `Extend` should always be an extends, otherwise something probably has gone horribly wrong.

Reply:
Yeah that's right. I can just probably return true from here. Same goes for `Component`.

Reply:
Yeah, you probably don't need to store the kind at all since all elements should now what they are.